### PR TITLE
Opportunity publish

### PR DESCRIPTION
**Pull request by S-unya**

### CODE REVIEW

File: .graphqlconfig.yml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 projects:
   ukri:
-    schemaPath: src/graphql/schema.json
```
Comment (torchiad) [inline .graphqlconfig.yml:3]: Is this expected?

File: amplify/backend/function/publishopportunitylisting/src/app.ts [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+
+AWS.config.region = "eu-west-1";
+
+export const getDBTableName = (env: string, apiId: string, type: string) =>
+    `${type}-${apiId}-${env}`;
+
+export const getSNSARNName = (
+    partition: string,
+    service: string,
+    region: string = "eu-west-1",
```
Comment (torchiad) [inline amplify/backend/function/publishopportunitylisting/src/app.ts]: We could use AWS.config.region here?

File: amplify/backend/function/publishopportunitylisting/src/api.d.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    description: string;
+    funders: string[];
+    fundersComplete?: boolean;
+    openDate?: string;
+    closeDate?: string;
+    websiteListings?: WebsiteListing[];
+    teammembers?: [TeamMember];
+    teammembersComplete?: boolean;
+    // Not sure what this is ... yet
+    type?: any;
```
Comment (torchiad) [inline amplify/backend/function/publishopportunitylisting/src/api.d.ts]: Change from "any" as we know the type now

File: amplify/backend/function/publishopportunitylisting/src/app.ts [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+// 3. add listing in format provided in full-event.json to SNS
+// 4. add `lastPublished` to listing in DynamoDb
+
+app.post("/opportunity-listing/publish", async (req, res) => {
+    try {
+        const { listingId, opportunityId } = req.body;
+        const env = process.env.env;
+        const apiId = api.ukri.output.GraphQLAPIIdOutput;
+
+        if (!env || !listingId) {
```
Comment (torchiad) [inline amplify/backend/function/publishopportunitylisting/src/app.ts]: to include opportunityId

File: amplify/backend/function/publishopportunitylisting/src/app.ts (hunk 2 above)
Comment (S-unya) [reply amplify/backend/function/publishopportunitylisting/src/app.ts]: done

File: amplify/backend/function/publishopportunitylisting/src/api.d.ts (hunk 3 above)
Comment (S-unya) [reply amplify/backend/function/publishopportunitylisting/src/api.d.ts]: Fixed a few things here

File: amplify/backend/function/publishopportunitylisting/src/app.ts (hunk 4 above)
Comment (S-unya) [reply amplify/backend/function/publishopportunitylisting/src/app.ts]: done

File: .graphqlconfig.yml (hunk 1 above)
Comment (S-unya) [reply .graphqlconfig.yml:3]: I don't know :(

**Comment by torchiad** [comment]
Looks good. @ascorbic Would you be happy for this to go into dev?

File: amplify/backend/function/publishopportunitylisting/src/app.ts [hunk 5]
```diff
@@ -0,0 +1,240 @@
+/*
+Copyright 2017 - 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (ascorbic) [inline amplify/backend/function/publishopportunitylisting/src/app.ts]: Can we remove this copyright notice? It's just for the boilerplate.

File: .graphqlconfig.yml (hunk 1 above)
Comment (ascorbic) [reply .graphqlconfig.yml:3]: No it needs fixing

**Comment by ascorbic** [comment]
Needs conflicts to be resolved and the environment fixed

**Pull request merged**